ui/src/utils/languageResolver.js

```diff
@@ -0,0 +1,18 @@
+import { map, isNull } from 'lodash';
+
+const getLabel = (name) => {
```

Comment:
I think we already have code for this somewhere, probably best to reuse that to avoid regressions.

Reply:
Yes, that code wasn't in a reusable state, this is.

Reply:
When you say it's not in a reusable state what do you mean? The code here looks very similar to `displayLangMap` from "ui/src/utils/xapi.js".

```js
export const displayLangMap = (langMap = new Map(), defaultDisplay = '') => {
  const langs = (navigator && navigator.languages) || [];
  const displays = langs
    .map(lang => langMap.get(lang))
    .filter(value => !!value);
  const display = displays[0] || langMap.first();
  return display || defaultDisplay;
};
```